Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -524,6 +526,32 @@ def run_retry_on_rate_limit(self, args_for_next_run: dict):
                               scheduled_command=ScheduledCommand(command=demisto.command(), next_run_in_seconds=60,
                                                                  args=args_for_next_run))
 
+    def handle_error_with_metrics(self, res):
+        self.create_api_metrics(res.status_code)
+        err_msg = 'Error in API call [{}] - {}' .format(res.status_code, res.reason)
```

Comment:
looks like this is a copy of what we have from CSP error handling. Ideally I would say to refactor that into a method and wrap it here
If we cant then lets a comment here of where this logic came from